ruma-events/src/lib.rs

```diff
@@ -256,6 +283,32 @@ impl RedactedSyncUnsigned {
     pub fn is_empty(&self) -> bool {
         self.redacted_because.is_none()
     }
+
+    /// Convert a `RedactedSyncUnsigned` into `RedactedUnsigned`, converting the
+    /// underlying sync redaction event to a full redaction event (with room_id).
+    pub fn into_full(self, room_id: RoomId) -> RedactedUnsigned {
+        match self.redacted_because.map(|b| *b) {
```

Comment:
Yeah, boxes are weird... That `|b| *b` even works is a compiler hack. But I still think this looks better than before.

Reply:
Yeah I agree destructuring the struct on the same level is way nicer.